### ISSUE TITLE
List view: Certain screen readers still switch modes when options button receives focus

### ISSUE THREAD

**Issue by alexstine** [WordPress/gutenberg#43729]
### Description
Whenever I open the list view, my screen reader remains in edit mode. This means all keyboard input gets sent to the application, Gutenberg. This is good because it allows me to navigate up and down the left side of the grid, or block names. The problem becomes when I enter the right column where the options buttons are, landing on a button forces screen readers from edit mode to browse mode. When a screen reader is in browse mode, you are experiencing the document with the virtual cursor. This means that up, down, left, and right arrows no longer navigate the grid.

### How to fix
I think adding `aria-hidden="true"` to the button may work but it will need testing. Essentially, the right side of the grid needs to have similar markup to the left side of the grid.

### Steps to reproduce
1. On Windows, start a free copy of [NVDA](https://nvaccess.org).
2. Open the editor.
3. Tab to the list view button in the top toolbar and press enter.
4. Notice how focus is placed in the left column.
5. Try navigating up and down, everything should work fine.
6. Try navigating to the right column. You will notice NVDA switches modes when it comes in contact with the options, button element.

**Comment by alexstine** [comment]
@andrewserong I remember this was a horrible pain to implement, but you did a great job! Up to take on this challenge?

**Comment by andrewserong** [comment]
Oh, this is a tricky one, thanks for the ping @alexstine! I'd love to help out with this, but unfortunately I'll be a bit busy these next couple of weeks as my priority at the moment is backporting changes to the Layout support for the 6.1 beta. Once those are sorted, I'm happy to have a go at this one if no-one beats me to it. Since it is a bug, we should be able to continue to work on it past the feature freeze for 6.1 if we don't have a fix in beforehand.

**Comment by alexstine** [comment]
Something tells me probably no one will get to it before you @andrewserong , but you never know. Even if this does not make 6.1, this will lay the ground work for future development I would like to see happen.

Thanks.

**Comment by talldan** [comment]
@alexstine I tried a fix in https://github.com/WordPress/gutenberg/pull/44291. Let me know what you think. I don't have a lot of experience with NVDA, so it's possible this may have other unwanted side effects, but I thought I'd try it out.

**Comment by talldan** [comment]
Another option is that we can completely remove the options dropdown menu from being part of the TreeGrid navigation and leave it with `tabIndex=-1` so that it's not navigable. I imagine that would have similar results to `aria-hidden`.
